File: internal/store/prompt_overrides.go
package store

import "database/sql"

// GetStage returns one pipeline stage's saved prompt override (content) and
// whether the stage is enabled. No row → ("", true): the compiled-in default
// prompt, stage on.
func (db *DB) GetStage(stage string) (content string, enabled bool, err error) {
	var en int
	err = db.QueryRow("SELECT content, enabled FROM prompt_overrides WHERE stage = ?", stage).Scan(&content, &en)
	if err == sql.ErrNoRows {
		return "", true, nil
	}
	return content, en != 0, err
}

// PutPromptOverride upserts a stage's prompt override (content). A new row
// defaults enabled=1; an existing row keeps its enabled flag.
func (db *DB) PutPromptOverride(stage, content string) error {
	const q = `
INSERT INTO prompt_overrides (stage, content, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(stage) DO UPDATE SET content = excluded.content, updated_at = CURRENT_TIMESTAMP`
	_, err := db.Exec(q, stage, content)
	return err
}

// SetStageEnabled upserts a stage's on/off flag, leaving any content override
// untouched (a new row defaults content=”).
func (db *DB) SetStageEnabled(stage string, enabled bool) error {
	en := 0
	if enabled {
		en = 1
	}
	const q = `
INSERT INTO prompt_overrides (stage, enabled, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(stage) DO UPDATE SET enabled = excluded.enabled, updated_at = CURRENT_TIMESTAMP`
	_, err := db.Exec(q, stage, en)
	return err
}

// ResetStageContent clears a stage's content override (reverting it to the
// compiled-in default prompt) while leaving its enabled flag in place.
func (db *DB) ResetStageContent(stage string) error {
	_, err := db.Exec(`UPDATE prompt_overrides SET content = '', updated_at = CURRENT_TIMESTAMP WHERE stage = ?`, stage)
	return err
}

File: scout/store/prompt_overrides.py
"""Per-stage outreach prompt overrides + on/off flags."""

from __future__ import annotations

import sqlite3


def get_stage(con: sqlite3.Connection, stage: str) -> tuple[str, bool]:
    """Return (content, enabled) for a stage's saved prompt override. No row →
    ("", True): the compiled-in default prompt, stage on."""
    row = con.execute(
        "SELECT content, enabled FROM prompt_overrides WHERE stage = ?", (stage,)
    ).fetchone()
    if row is None:
        return "", True
    return row[0], row[1] != 0


def put_prompt_override(con: sqlite3.Connection, stage: str, content: str) -> None:
    """Upsert a stage's prompt override (content). A new row defaults enabled=1."""
    con.execute(
        """
INSERT INTO prompt_overrides (stage, content, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(stage) DO UPDATE SET content = excluded.content, updated_at = CURRENT_TIMESTAMP""",
        (stage, content),
    )


def set_stage_enabled(con: sqlite3.Connection, stage: str, enabled: bool) -> None:
    """Upsert a stage's on/off flag, leaving any content override untouched."""
    con.execute(
        """
INSERT INTO prompt_overrides (stage, enabled, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(stage) DO UPDATE SET enabled = excluded.enabled, updated_at = CURRENT_TIMESTAMP""",
        (stage, 1 if enabled else 0),
    )


def reset_stage_content(con: sqlite3.Connection, stage: str) -> None:
    """Clear a stage's content override (revert to the compiled-in default),
    leaving its enabled flag in place."""
    con.execute(
        "UPDATE prompt_overrides SET content = '', updated_at = CURRENT_TIMESTAMP WHERE stage = ?",
        (stage,),
    )
